lib/api.js

```diff
@@ -0,0 +1,68 @@
+'use strict'; // eslint-disable-line strict
+
+const Config = require('./Config.js');
+const Logger = require('./Logger.js');
+
+class API {
+
+    /**
+     * This is the constructor of the Logger class. It takes optional
+     * configuration parameters, that allow to modify its behavior.
+     *
+     * @param {object} config         - A configuration object for werelogs.
+     * @param {string} config.level   - The string name of the logging level
+     *                                  ('trace', 'debug', 'info', 'warn',
+     *                                  'error' and 'fatal' in order of
+     *                                  importance.)
+     * @param {string} config.dump    - The string name of the log dumping level
+     *                                  ('trace', 'debug', 'info', 'warn',
+     *                                  'error' and 'fatal' in order of
+     *                                  importance.)
+     * @param {object[]} config.streams - The array of streams into which to
+     *                                    log. This is an Array of objects
+     *                                    which have a field named 'stream',
+     *                                    which is writeable.
+     *
+     * @returns {undefined}
+     */
+    constructor(config) {
+        this.config = new Config(config);
+        this.preboundLogger = Logger.bind(null, this.config);
+    }
+
+    /**
+     * This is a thunk function that allows reconfiguring the streams and log
+     * levels of all Logger and future RequestLogger objects. Note that
+     * existing RequestLogger will live their lifespan retaining the old
+     * configuration.
+     * If the provided configuration is erroneous, the function may throw
```

Comment:
Took the opportunity to provide one more possible error type; and update related documentation.